libr/anal/esil.c

```diff
@@ -2309,19 +2248,14 @@ static int esil_mem_diveq_n(RAnalEsil *esil, int bits) {
 	return ret;
 }
 
-static int esil_mem_diveq1(RAnalEsil *esil) {
-	return esil_mem_diveq_n (esil, 8);
-}
-static int esil_mem_diveq2(RAnalEsil *esil) {
-	return esil_mem_diveq_n (esil, 16);
-}
-static int esil_mem_diveq4(RAnalEsil *esil) {
-	return esil_mem_diveq_n (esil, 32);
-}
-static int esil_mem_diveq8(RAnalEsil *esil) {
-	return esil_mem_diveq_n (esil, 64);
+static int esil_mem_diveq(RAnalEsil *esil, const char *word) {
+	int bytes;
+	if (sscanf (word, "/=[%d]", &bytes) != 1) {
```

Comment:
not efficient

Reply:
Performance is not relevant here. If you do care about performance, don't do string manipulation as the file currently does. The command execution is already very inefficient anyway.

You always care about things that do not particularly matter and this is very frustrating.

Reply:
Performance is **relevant** in ESIL emulation, doing sscanf to get a number when you already know the size is completely unnecessary. And I would say the same thing for most of @MaskRay and @ret2libc contributions that are mainly things that i completely irrelevant to 99% of the users, but i have to live with that.

Command execution is completely out of the scope of this PR, and it's a well known thing, but the solution is not easy, otherwise someone would propose a magic solution.

About frustration, i'm already really frustrated about spending most of my day reviewing and caring about PRs that nobody asked for them and managing discussions of egos without having a real perspective of the needed problems or needs of the project.

But i'm used to live under stress and frustration, so welcome to r2land.

Reply:
you can use rnum, or just a simple switch

Reply:
Performance is relevant imo, but only to a certain degree. We don't want hyper complitcated perf optimized esil, which gets even harder to maintain. But there is no reason to use sscanf. Esil affects some kinds of analysis directly, when it comes to perf.

Reply:
@radare "but the solution is not easy" Some of the design gets in the away and some contribution willing to improve the things can easily be frowned upon. It is marvelous that you devote so much time on it (and I admire you much for doing this).. but sometimes because of the stress maintainers are making decisions that deviate from usually software development much, though that may attribute to the characteristics of reverse engineers.

Reply:
@condret the code (esil.c) has become very complicated and very difficult to improve. This is not what a conceptually simple stack machine ought to be. Things should be simplified, then improved. But you are oftentimes emphasizing on things that don't matter much.